cupy/lib/polynomial.pyx

```diff
@@ -4,6 +4,41 @@ import cupy
 from cupy.core.core cimport ndarray
 
 
+def polysub(a1, a2):
+    """Computes the difference of two polynomials.
+
+    Args:
+        a1 (scalar, cupy.ndarray or cupy.poly1d): first input polynomial.
+        a2 (scalar, cupy.ndarray or cupy.poly1d): second input polynomial.
+
+    Returns:
+        cupy.ndarray or cupy.poly1d: The difference of the inputs.
+
+    .. seealso:: :func:`numpy.polysub`
+
+    """
+    truepoly = False
+    if isinstance(a1, poly1d):
+        a1 = a1._coeffs
+        truepoly = True
+    if isinstance(a2, poly1d):
+        a2 = a2._coeffs
+        truepoly = True
+    a1 = cupy.atleast_1d(a1)
+    a2 = cupy.atleast_1d(a2)
+    if a1.shape[0] <= a2.shape[0]:
+        val = cupy.pad(a1, (a2.shape[0] - a1.shape[0], 0))
+        val = val.astype(cupy.result_type(a1, a2), copy=False)
+        val -= a2
+    else:
+        val = cupy.pad(a2, (a1.shape[0] - a2.shape[0], 0))
+        val = val.astype(cupy.result_type(a1, a2), copy=False)
+        val -= 2 * val - a1
```

Comment:
https://github.com/cupy/cupy/pull/3593/commits/1092ef2d548ac9faeefc2194ea6f9d3198cadca5 is simpler, but I modified it to reuse memory